src/main/java/com/shaft/cucumber/ElementSteps.java

```diff
@@ -1,71 +1,76 @@
 package com.shaft.cucumber;
 
-import com.shaft.gui.browser.BrowserActions;
-import com.shaft.gui.browser.BrowserFactory;
-import com.shaft.gui.element.ElementActions;
 import com.shaft.validation.Assertions;
 import io.cucumber.java.ParameterType;
-import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import io.cucumber.java.en.When;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebDriver;
 
 import java.util.Arrays;
 
-public class GuiActionSteps {
-    //TODO: handle shaft_engine listeners
-    private final ThreadLocal<WebDriver> driver = new ThreadLocal<>();
+public class ElementSteps {
+    private final ThreadLocal<WebDriver> driver;
 
-    @Given("I Open the target browser")
-    public void BrowserFactory_getBrowser() {
-        driver.set(BrowserFactory.getBrowser());
-    }
-
-    @When("I Navigate to {string}")
-    public void BrowserActions_navigateToURL(String url) {
-        BrowserActions.navigateToURL(driver.get(), url);
-    }
-
-    @When("I Navigate to {string} and get redirected to {string}")
-    public void BrowserActions_navigateToURL(String url, String urlAfetrRedirection) {
-        BrowserActions.navigateToURL(driver.get(), url, urlAfetrRedirection);
-    }
-
-    @When("I Close the current window")
-    public void BrowserActions_closeCurrentWindow() {
-        BrowserActions.closeCurrentWindow(driver.get());
+    public ElementSteps(ThreadLocal<WebDriver> driver) {
```

Comment:
Codacy found an issue: [Avoid reassigning parameters such as 'driver'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5731302)

Reply:
that's part of the dependency injection implementation